### PR TITLE
Fixed MultipleAvatars

### LINKED ISSUE #1064: Display avatars in multi-participants chats
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___
View the job on Upwork [here](https://www.upwork.com/jobs/~01728febe6495dbae6). 
**Platform - version:**  
Build 280, all platforms 

**Steps to view existing functionality which should be changed (see attached Image 1)**

1. Open the app, type in your username and password, and tap 'Log in'
2. Once you are logged in begin a chat with 2 or more additional participants. Do this by clicking typing a name in the “Find or start chat” field and clicking “Add” next to each participant.
3. Notice that there are no avatars displayed for the chat participants.

**Desired Result:**
1. When a chat has 3 participants, show the avatars of the 2 other participants (see attached Image2)
2. When a chat has 4 or more participants, show the avatar of the user whose username comes first alphabetically and show a circular counter for the number of chat participants. Example: 4 participants (including the viewing user) would show as one avatar and 3 shown in the circular counter (see attached Image3).

------
# Photos
 
## Image 1 - Existing functionality: 

image

## Image 2: 3 participants, show the avatars of the 2 other participants 

image

## Image 3: 4+ participants, show the avatar of the first user and circular counter 

image

### PR DESCRIPTION
**Pull request by whitesnow1120**
Please review my update. @roryabraham @marcaaron @shawnborton 

### Details

- Updated code style
- Fixed line-height of avatar text

### Fixed Issues

Fixes [1064](https://github.com/Expensify/Expensify.cash/issues/1064)

### Tests

Once you are logged in, you can chat with other participants (2+).
When a chat has 2 participants (me and another participant), showed the single avatar (40X40) of another user.
When a chat has 3 participants, showed the avatars (24X24) of the 2 other participants.
When a chat has 4 or more participants, showed the avatar (24X24) of the user whose username comes first alphabetically, and showed a circular (24X24) counter for the number of chat participants.
The font size of number is 11 (variables.fontSizeSmall) and background color is themeColors.textReversed.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

## iOS
iOS

## Android
android

## Web
web

## mobile web
mobile web

## mac desktop
desktop

### CODE REVIEW

File: src/styles/styles.js [hunk 1]
```diff
@@ -755,32 +755,32 @@ const styles = {
...
 
-    chatSwitcherSingleAvatar: {
+    singleAvatar: {
         height: variables.avatarSizeNormal,
         width: variables.avatarSizeNormal,
         borderRadius: variables.avatarSizeNormal,
     },
 
-    chatSwitcherBigAvatar: {
+    bigAvatar: {
```
Comment (shawnborton) [inline src/styles/styles.js:764]: I don't know if I like the name `bigAvatar` - this is actually just the standard, normal size avatar we'd expect to see across the application and not necessarily "big"

**Comment by shawnborton** [comment]
So one thing I am noticing is that you are defining `avatarSizeNormal` as being the avatar size that is 24 by 24. In reality, the normal avatar size is actually just the standard `componentSizeNormal` size of 40 by 40. 

The small avatar size of 24 x 24 is kind of a one-off value at the moment. We've defined `componentSizeSmall` as 28 by 28, which you'll notice that is larger than 24x24. 

So I think perhaps we shouldn't add the avatar size of 24 as a global variable just yet until that value is being used as more of a global variable in other pieces of UI. Thoughts @marcaaron and @roryabraham ?

File: src/styles/styles.js (hunk 1 above)
Comment (whitesnow1120) [reply src/styles/styles.js:764]: How about `normalAvatar `instead of `bigAvatar`?

Comment (shawnborton) [reply src/styles/styles.js:764]: I would just switch the order to be `avatarNormal`

File: src/pages/home/sidebar/ChatSwitcherView.js [hunk 2]
```diff
@@ -428,7 +428,7 @@ class ChatSwitcherView extends React.Component {
                 alternateText: personalDetail.login,
                 searchText: personalDetail.displayName === personalDetail.login ? personalDetail.login
                     : `${personalDetail.displayName} ${personalDetail.login}`,
-                icon: Array.isArray(personalDetail.avatarURL) ? personalDetail.avatarURL : [personalDetail.avatarURL],
+                icons: _.isArray(personalDetail.avatarURL) ? personalDetail.avatarURL : [personalDetail.avatarURL],
```
Comment (marcaaron) [inline src/pages/home/sidebar/ChatSwitcherView.js:431]: Still unclear why we need the check? When is this going to be an array?

File: src/styles/styles.js (hunk 1 above)
Comment (whitesnow1120) [reply src/styles/styles.js:764]: avatarNormal is already declared.

Comment (shawnborton) [reply src/styles/styles.js:764]: Is there any reason why we can't just reuse it then? Meaning, we don't need to write new styles, just reuse that same component that already exists.

Comment (whitesnow1120) [reply src/styles/styles.js:764]: Ok.

Comment (marcaaron) [reply src/styles/styles.js:764]: +1

**Review by marcaaron — CHANGES_REQUESTED** [review]
Left another question here.

File: src/pages/home/sidebar/ChatSwitcherView.js (hunk 2 above)
Comment (whitesnow1120) [reply src/pages/home/sidebar/ChatSwitcherView.js:431]: Yes. avatarURL is a string, not array.
I don't need to check it.

**Comment by marcaaron** [comment]
>So I think perhaps we shouldn't add the avatar size of 24 as a global variable just yet until that value is being used as more of a global variable in other pieces of UI. Thoughts @marcaaron and @roryabraham ?

That makes sense to me. Let's wait for that to become "a thing" before making it a variable maybe?

**Comment by shawnborton** [comment]
Works for me!

**Review by marcaaron — CHANGES_REQUESTED** [review]
@shawnborton Not sure if you are seeing this but when I went to create a new group DM between myself and 3 others the avatar has the number `3`. Is it not supposed to say how many additional users that are not the one user shown and the current user? I would expect this to say `2` and not `3`.

Screen Shot 2021-01-04 at 4 12 23 PM

**Comment by shawnborton** [comment]
Great catch, I agree with you there @marcaaron

**Comment by whitesnow1120** [comment]
I removed a global variable (avatarSizeNormal) and changed the number of participants.

**Review by shawnborton — APPROVED** [review]

**Review by roryabraham — APPROVED** [review]
LGTM

**Review by marcaaron — APPROVED** [review]

**Pull request merged**